### ISSUE TITLE
einsum optimize overrides order

### ISSUE THREAD

**Issue by bmcfee** [numpy/numpy#16415]
I'm not sure if this is intended / expected behavior, but changing the `optimize=` parameter in `einsum` results in different memory layouts, even when the `order=` parameter is given.

This to me seems to contradict the documentation for the order parameter.

I didn't see anything else in the documentation to suggest this behavior, and none of the open issues on einsum seem to cover this (though I could have missed one).

### Reproducing code example:

```python
import numpy as np

A = np.ones((2, 5, 10), order='F')
B = np.ones((8, 5), order='F')
print(A.flags)  # verify that both inputs are F-contiguous
print(B.flags)
C = np.einsum('...ft,mf->...mt', A, B, order='A', optimize=False)
print(C.flags)  # result is F-contiguous
C2 = np.einsum('...ft,mf->...mt', A, B, order='A', optimize='greedy')
print(C2.flags)  # result is not F-contiguous (or C-contiguous)

```

### Error message:

n/a ; no error, but results are not as expected.

### Numpy/Python version information:

Output from 'import sys, numpy; print(numpy.__version__, sys.version)'

```
1.18.1 3.7.7 (default, May  7 2020, 21:25:33) 
[GCC 7.3.0]
```

**Comment by seberg** [comment]
The optimized path can go into BLAS which is much faster and probably simply ignores the order.  Not sure how easy it is to fix, unless we simply skip the optimization.
Its not unlikely that the fast path creates the output array and passes it into `np.dot` basically, in which case it may be actually be fairly straight forward to fix.

@dgasmith just a gentle ping in case you want to/have time to look at it.

**Comment by dgasmith** [comment]
The optimized output is a view to a `...tm` array as it is the cheapest way to align the GEMMs not the `...mt` requested ordering which is why the output is neither C or F ordering with optimization on.

It doesn't look like we handle the GEMM cases well with regard to out ordering well. This should be a fairly easy add for C/F/A ordering, it has been a bit but I think `K` ordering for several tensors is a "whatever comes out". I'll double check the C code, but let me know if I'm wrong here.

I will try to get a patch in this weekend.

**Comment by seberg** [comment]
K means keep order, but OTOH, the main reason for it is to order as fast as possible given the input. Maybe for K we can just document briefly in some notes.

**Comment by rossbar** [comment]
AFAICT this was fixed in #16446. If there are still components that need updating (perhaps updating the docstring for the `order=K` case), please feel free to reopen/open a new issue!
